XpringKit/Utils.swift

```diff
@@ -1,66 +1,132 @@
+    ///   - isTest Whether the address is for use on a test network, defaults to `false`.
+    /// - Returns: A new x-address if inputs were valid, otherwise undefined.
+    public static func encode(classicAddress: Address, tag: UInt32? = nil, isTest: Bool = false) -> Address? {
+        return javaScriptUtils.encode(classicAddress: classicAddress, tag: tag, isTest: isTest)
+    }
 
-  /// Decode a classic address from a given x-address.
-  ///
-  /// - SeeAlso: https://xrpaddress.info/
-  ///
-  /// - Parameter xAddress: The xAddress to decode.
-  /// - Returns: a tuple containing the decoded address,  tag and bool indicating if the address was on a test network.
-  public static func decode(xAddress: Address) -> (classicAddress: String, tag: UInt32?, isTest: Bool)? {
-    return javaScriptUtils.decode(xAddress: xAddress)
-  }
+    /// Decode a classic address from a given x-address.
+    ///
+    /// - SeeAlso: https://xrpaddress.info/
+    ///
+    /// - Parameter xAddress: The xAddress to decode.
+    /// - Returns: a tuple containing the decoded address,  tag and bool indicating if the address was on a test network.
+    public static func decode(xAddress: Address) -> (classicAddress: String, tag: UInt32?, isTest: Bool)? {
+        return javaScriptUtils.decode(xAddress: xAddress)
+    }
 
-  /// Convert the given transaction blob to a transaction hash.
-  ///
-  /// - Parameter transactionBlobHex: A hexadecimal encoded transaction blob.
-  /// - Returns: A hex encoded hash if the input was valid, otherwise undefined.
-  public static func toTransactionHash(transactionBlobHex: Hex) -> Hex? {
-    return javaScriptUtils.toTransactionHash(transactionBlobHex: transactionBlobHex)
-  }
+    /// Convert the given transaction blob to a transaction hash.
+    ///
+    /// - Parameter transactionBlobHex: A hexadecimal encoded transaction blob.
+    /// - Returns: A hex encoded hash if the input was valid, otherwise undefined.
+    public static func toTransactionHash(transactionBlobHex: Hex) -> Hex? {
+        return javaScriptUtils.toTransactionHash(transactionBlobHex: transactionBlobHex)
+    }
+
+    /// Convert the given string to a bytes array UInt8.
+    ///
+    /// - Parameter hex: A string.
+    /// - Returns: A byte array in [UInt8].
+    public static func toByteArray(hex: String ) -> [UInt8] {
+        // remove "-" from Hexadecimal
+        let hexString = hex.removeWord( "-" )
+        let size = hexString.count / 2
+        var result: [UInt8] = [UInt8]( repeating: 0, count: size ) // array with length = size
+        for i in stride( from: 0, to: hexString.count, by: 2 ) {
+            let subHexStr = hexString.subString( i, length: 2 )
+            guard let _uint8 = UInt8( subHexStr, radix: 16 ) else {
+                return []
+            }
+            result[ i / 2 ] = _uint8
+        }
+        return result
+    }
+}
+extension String {
+
+    func subString( _ from: Int, length: Int ) -> String {
```

Comment:
I'd prefer we just use `prefix` to get the first two elements, rather than re-implementing a `substring` method:
https://developer.apple.com/documentation/swift/string/2894830-prefix

This method is essentially implementing [String's subscript](https://developer.apple.com/documentation/swift/string/2943910-subscript) method. Would prefer we use the official definitions provided by foundation rather than hand rolling our own.

Reply:
I couldn't get the prefix to work, It seems its a custom function. You're welcome to refactor the function.